src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -96,13 +96,21 @@ static SQLitePersistentStorage()
         {
             // Attempt to load the correct version of e_sqlite.dll.  That way when we call
             // into SQLitePCL.Batteries_V2.Init it will be able to find it.
-            var myFolder = Path.GetDirectoryName(
-                typeof(SQLitePersistentStorage).Assembly.Location);
+            //
+            // Only do this on Windows when we can safely do the LoadLibrary call to this
+            // direct dll.  On other platforms, it is the responsibility of the host to ensure
+            // that the necessary sqlite library has already been loaded such that SQLitePCL.Batteries_V2
+            // will be able to call into it.
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                var myFolder = Path.GetDirectoryName(
+                    typeof(SQLitePersistentStorage).Assembly.Location);
 
-            var is64 = IntPtr.Size == 8;
-            var subfolder = is64 ? "x64" : "x86";
+                var is64 = IntPtr.Size == 8;
+                var subfolder = is64 ? "x64" : "x86";
 
-            LoadLibrary(Path.Combine(myFolder, subfolder, "e_sqlite3.dll"));
+                LoadLibrary(Path.Combine(myFolder, subfolder, "e_sqlite3.dll"));
+            }
 
             // Necessary to initialize SQLitePCL.
             SQLitePCL.Batteries_V2.Init();
```

Comment:
❔ This has a pretty horrific behavior if it throws an exception. Can/should we capture the exception details for a possible failure scenario described by the conditional block above, especially considering we are able to sanely fall back to the NOP storage if it fails or is not available?